radicle/src/storage/git/transport/local.rs

```diff
@@ -1,2 +1,107 @@
+//! The local git transport protocol.
 pub mod url;
 pub use url::{Url, UrlError};
+
+use std::cell::RefCell;
+use std::process;
+use std::str::FromStr;
+use std::sync::Once;
+
+use once_cell::sync::OnceCell;
+
+use crate::storage;
+use crate::storage::git::Storage;
+
+use super::ChildStream;
+
+thread_local! {
+    /// Stores a storage instance per thread.
+    static THREAD_STORAGE: OnceCell<Storage> = OnceCell::default();
```

Comment:
The comment may be more helpful with a reason for it.   My understanding is its at most one per thread to avoid race conditions.